### PR TITLE
Handle missing export error

### LINKED ISSUE #104: Plugin incompatible with Node.js `exports` package.json field
This plugin tries to resolve the `package.json` files of processed modules: https://github.com/sveltejs/rollup-plugin-svelte/blob/85d25fbc78fab486373849548afddf6a80db8a95/index.js#L177-L180

This breaks with npm modules that make use of the [new `exports` field](https://nodejs.org/api/esm.html#esm_package_entry_points) but do not specify a subpath export for the `package.json` file.

This issue surfaced in https://github.com/uuidjs/uuid/issues/444

IMO this plugin can no longer expect that every npm module exports the `package.json` file and the resolver logic has to be adjusted accordingly.

Maybe in https://github.com/sveltejs/rollup-plugin-svelte/blob/85d25fbc78fab486373849548afddf6a80db8a95/index.js#L48-L55

we could add `ERR_PACKAGE_PATH_NOT_EXPORTED` as an "acceptable" error as well?

### PR DESCRIPTION
**Pull request by pngwn**
Fixes #104 

Alternative to to #108 #109 

This PR gathers all of these error together, ignoring duplicates, and reports them at the end with a helpful message. The other PRs were very noisy and resulted in many duplicates being posted to the console. 

Screenshot 2020-07-31 at 22 04 32

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -334,7 +341,13 @@ module.exports = function svelte(options = {}) {
 				}, this.warn);
 
 				css(writer);
+
+				
 			}
+			if (pkg_export_errors.size < 1) return;
+
+			console.warn('\nrollup-plugin-svelte: The following packages did not export their `package.json` file so we could not check the `svelte` field. If you had difficulties importing svelte components from a package, please contact the author and ask them to export their package.json file.\n');
```
Comment (pngwn) [inline index.js:349]:
```suggestion
			console.warn('\nrollup-plugin-svelte: The following packages did not export their `package.json` file so we could not check the `svelte` field. If you had difficulties importing svelte components from a package, then please contact the author and ask them to export the package.json file.\n');
```

File: index.js [hunk 2]
```diff
@@ -50,6 +51,12 @@ function tryResolve(pkg, importer) {
 		return relative.resolve(pkg, importer);
 	} catch (err) {
 		if (err.code === 'MODULE_NOT_FOUND') return null;
+		if (err.code === 'ERR_PACKAGE_PATH_NOT_EXPORTED') {
+			if (!pkg_export_errors.has(pkg)) {
```
Comment (lukeed) [inline index.js:55]: No need to check has(). Just add it and Set will dedupe

File: index.js [hunk 3]
```diff
@@ -50,6 +51,12 @@ function tryResolve(pkg, importer) {
...
+				pkg_export_errors.add(pkg.replace(/\/package.json$/, ''));
+			}
+			return null;
```
Comment (pngwn) [inline index.js:58]:
```suggestion
		if (err.code === 'ERR_PACKAGE_PATH_NOT_EXPORTED') {
			pkg_export_errors.add(pkg.replace(/\/package.json$/, ''));
			return null;
```

File: index.js (hunk 2 above)
Comment (pngwn) [reply index.js:55]: nice

**Comment by lukeed** [comment]
This is ready to go

**Pull request merged**